packages/core-modules/contracts/modules/CoreElectionModule.sol

```diff
@@ -131,26 +136,55 @@ contract CoreElectionModule is IElectionModule, ElectionStorage, OwnableMixin {
         _electionStore().nextNominationPeriodPercent = percent;
     }
 
-    function elect(address[] memory candidates) external view override {
-        uint seatCount = _electionStore().nextSeatCount;
+    function elect(address[] memory candidates) external override {
+        ElectionStore storage store = _electionStore();
+        ElectionData storage electionData = store.electionData;
 
-        if (candidates.length != seatCount) {
+        if (candidates.length == 0) {
             revert InvalidCandidatesCount();
         }
 
-        // TODO: if msg.sender already voted, rollback previous votes.
+        if (electionData.addressVoted[msg.sender]) {
+            // TODO: if msg.sender already voted, rollback previous votes and allow to re-vote
+            revert AlreadyVoted();
+        }
 
-        // TODO: Assign votes to each address (same to all)
-        // uint votePower = ERC20(_electionStore().electionTokenAddress).balanceOf(msg.sender);
+        // The voting power is the amount of votes we are going to assign to the given candidates
+        uint votePower = ERC20(store.electionTokenAddress).balanceOf(msg.sender);
+
+        for (uint i = 0; i < candidates.length; i++) {
+            address candidate = candidates[i];
+
+            // Check that the candidate is a nominee
+            if (electionData.nomineeIndexes[candidate] == 0) {
+                revert InvalidCandidate(candidate);
+            }
+
+            // Check that all the values on the candidates Array are unique
+            if (i < candidates.length - 1) {
+                for (uint256 j = i + 1; j < candidates.length; j++) {
+                    address nextCandidate = candidates[j];
+                    if (candidate == nextCandidate) {
+                        revert InvalidCandidateRepeat(candidate);
+                    }
+                }
+            }
+
+            // Assign votes to the given candidate
+            electionData.nomineeVotes[candidate] += votePower;
+        }
+
+        // Save vote data for future usage
+        VoteData memory voteData = VoteData({candidates: candidates, votePower: votePower});
+        electionData.votes.push(voteData);
+        electionData.votesIndexes[msg.sender] = electionData.votes.length - 1;
```

Comment:
Arguable, but to me `voteIndices` sounds more natural.

Reply:
Also arguable, but going to keep `index`, because is the same notation that is used on the [official docs](https://docs.soliditylang.org/en/develop/types.html?highlight=array#iterable-mappings).